src/Screens/Checkout/Checkout.js

```diff
@@ -0,0 +1,338 @@
+import "./checkout.css";
+import AddressForm from "./../../Components/AddressForm/AddressForm";
+import { useEffect, useState } from "react";
+import uuid from "react-uuid";
+import { useProducts } from "./../../Context/products-context.js";
+import Empty from "./../../Components/Empty/Empty";
+import emptyImage from "./../../Media/empty-cart.png";
+import { useToast } from "./../../Context/toast-context";
+import { useAddress } from "./../../Context/address-context";
+import Navbar from "./../../Components/Navbar/Navbar";
+
+function Checkout() {
+  const { cartArray } = useProducts();
+  var storedSavedAddresses = JSON.parse(
+    localStorage.getItem("SAVED_ADDRESSES")
+  );
+
+  if (storedSavedAddresses === null) storedSavedAddresses = [];
+
+  const cartTotal = cartArray.reduce((accumulator, currentValue) => {
+    accumulator += currentValue.bookPrice * currentValue.bookQuantity;
+    return accumulator;
+  }, 0);
+
+  const cartQuantity = cartArray.reduce((accumulator, currentValue) => {
+    accumulator += currentValue.bookQuantity;
+    return accumulator;
+  }, 0);
+
+  var shippingTotal = 25 + cartQuantity * 25;
+
+  if (storedSavedAddresses === undefined) storedSavedAddresses = [];
+
+  const { state, dispatch } = useAddress();
+  const { toggleToast, toastVisibility, toastColor, toastText } = useToast();
+
+  const [savedAddresses, setSavedAddresses] = useState(storedSavedAddresses);
+  const [formDisplay, setFormDisplay] = useState(false);
+  const [editingAddress, setEditingAddress] = useState(false);
+  const [editedAddressId, setEditedAddressId] = useState("");
+
+  const [selectedAddress, setSelectedAddress] = useState({
+    name: "",
```

Comment:
it is not a big problem and I have seen people implementing this but honestly, I am not a fan of using states as objects. This is just a little Opinionated suggestion.